lib/aggregate/attribute/string.rb

```diff
@@ -1,17 +1,53 @@
 class Aggregate::Attribute::String < Aggregate::Attribute::Builtin
 
+  class Aggregate::EncryptionError < StandardError; end;
+
   def self.available_options
     Aggregate::Attribute::Builtin.available_options + [
       :size, # The maximum length of the string
+      :encrypted, # Whether or not we need to encrypt a field
     ]
   end
 
   def load(value)
-    value.to_s
+    options[:encrypted] ? decrypt(value) : value.to_s
+  end
+
+  def decrypt(value)
+    # get string, convert to hash, read decoded iv, decode value, decrypt with iv and value
+    hash = ActiveSupport::JSON.decode value
+
+    find_decrypted_value(Base64.urlsafe_decode64(hash["encrypted_data"]), Base64.urlsafe_decode64(hash["initilization_vector"]))
+  end
+
+  def find_decrypted_value(value, iv)
+    encrypted_value = nil
+    Aggregate::Base.hashed_keys.each do |k, v|
+      begin
+        encrypted_value = Encryptor.decrypt(value: value, key: v, iv: iv)
+      rescue OpenSSL::Cipher::CipherError
+        nil
+      end
+    end
+    encrypted_value.presence or raise Aggregate::EncryptionError, "correct key not found"
+    encrypted_value
+  end
+
+  def encrypt(value)
+    Aggregate::Base.hashed_keys.presence or raise Aggregate::EncryptionError, "must specify a key"
+
+    # Generate random iv, store as hash, encode into JSON safe string
+    iv = SecureRandom.random_bytes(12)
+
+    encrypted_data = Encryptor.encrypt(value: value, key: Aggregate::Base.hashed_keys.first.last, iv: iv)
```

Comment:
The idea here, was that I generalized the gem for future use without much effort at all. You can either pass in a key (as a string) or a list of keys (I am going to fix it to a list today).